### PR TITLE
monikers of toc file is the intersection if monikerRange setting in the config and file metadata

### PR DESCRIPTION
**Pull request by 928PJY**
#3607

### CODE REVIEW

File: src/Microsoft.Docs.Template/Models/TableOfContents.cs [hunk 1]
```diff
@@ -20,8 +20,16 @@ public class TableOfContentsMetadata
     {
         public List<string> Monikers { get; set; }
 
+        [JsonProperty(PropertyName = "monikerRange")]
+        public string MonikerRange { get; set; }
+
         [JsonExtensionData]
         public JObject ExtensionData { get; set; }
+
+        public bool ShouldSerializeMonikerRange()
```
Comment (yufeih) [inline src/Microsoft.Docs.Template/Models/TableOfContents.cs]: what is this for?

Comment (928PJY) [reply src/Microsoft.Docs.Template/Models/TableOfContents.cs]: to ignore monikerRange when serialization

File: src/docfx/build/toc/BuildTableOfContents.cs [hunk 2]
```diff
@@ -27,20 +27,21 @@ internal static class BuildTableOfContents
...
+            errors.AddIfNotNull(monikerError);
 
             var model = new TableOfContentsModel
             {
                 Items = tocModel,
                 Metadata = metadata,
             };
 
-            return (errors, model, monikers);
+            return (errors, model, metadata.Monikers);
```
Comment (OsmondJiang) [inline src/docfx/build/toc/BuildTableOfContents.cs]: >Monikers [](start = 44, length = 8)

will this moniker seeting has impacts to toc nodes' moniker

File: docs/designs/conceptual-versioning.md [hunk 3]
```diff
@@ -514,10 +512,12 @@ In current docfx v3, Toc file is build at the same time with `page` files, becau
 
 When build the toc file(including the toc file included in another), we have to cover:
 
-1. *Monikers* of every node in toc. We don't support setting monikers of node in toc file, 
+1. *Monikers* of every node in toc(We don't support setting monikers of node in toc file).
     1. If the node is a text node(`## Header`), the *monikers* of this node is the collection of its children's monikers.
     1. If the node is a link node(`## [Header](a.md)`), the *monikers* of this node is the collection of its children's monikers and the monikers of this node.
-1. *Monikers* of toc file. We don't support setting monikers of the toc file by config or file metadata, the *monikers* of toc file is the collection of every node's monikers.
+1. *Monikers* of toc file - Intersection of the monikerRange setting in the config and file metadata(If the intersection is empty, a warning will be logged).
```
Comment (OsmondJiang) [inline docs/designs/conceptual-versioning.md]: do you need add the case we talked offline?

Comment (928PJY) [reply docs/designs/conceptual-versioning.md]: the test inside `moniker.yml` have covered this?

---
In reply to: [239359268](https://github.com/dotnet/docfx/pull/3725#discussion_r239359268) [](ancestors = 239359268)

**Pull request merged**